6-Typescript/server/tests/todo.test.ts

```diff
@@ -0,0 +1,126 @@
+
+import { AxiosError } from 'axios'
+import { Testkit } from './env/testkit'
+import { Item } from '../interfaces/Item'
+import { AppDriver } from './drivers/appDriver'
+import { DBDriver } from './drivers/DBDriver'
+import { MongoDBController } from '../db/mongoDBController'
+import { v4 as uuid } from 'uuid'
+import { Guid } from '../interfaces/Guid'
+
+const db = new MongoDBController()
+const appDriver: AppDriver = new AppDriver('http://localhost:3000')
+const dbDriver: DBDriver = new DBDriver(db)
+const testkit: Testkit = new Testkit(appDriver, dbDriver)
+
+const userA: Guid = appDriver.userID
+const userB: Guid = uuid()
+
+const existingItemID: Guid = uuid()
+const nonExistingItemID: Guid = uuid()
+
+
+describe("testing the app's endpoints'", () => {
+
+    beforeAll(async () => await testkit.app.start(db))
+    beforeEach(async () => await testkit.before())
+    afterEach(async () => await testkit.after())
+    afterAll(async () => await testkit.app.close())
+    
+    it("should get an empty list from a user with no items", async () => {
+        //given - empty DB
+        //when
+        const itemsFromDB: Item[] = await testkit.app.getAll()
+        //then
+        expect(itemsFromDB.length).toBe(0)
+    })
+    it("should get all items from a user", async () => {
+        //given
+        const itemsToDB = [testkit.anItem({ userID: userA }), testkit.anItem({ userID: userA }), testkit.anItem({ userID: userB })]
+        await testkit.db.setItemInDB(itemsToDB[0].userID, itemsToDB[0]._id, itemsToDB[0].title, itemsToDB[0].content);
```

Comment:
did u see this comment? I can't see the fix

Reply:
the setItemsInDB function now receives an Item type argument only. there's no longer a version that receives _id, title and so on in the entire code, i don't now why this comment show old code (I just pulled the code from remote, this comment shows old code). I decided not to make a version that receives an array of Items because it wouldn't have an actual use in production